sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileWriteStream.cs

```diff
@@ -59,7 +62,7 @@ protected override async Task FlushInternal(bool async, CancellationToken cancel
             Response<PathInfo> response = await _fileClient.FlushInternal(
                 position: _writeIndex,
                 retainUncommittedData: default,
-                close: default,
+                close: _closeEvent,
```

Comment:
should we throw is someone is attempting to flush again and closeEvent = true ?

Reply:
I think we should let the service handle this, I'm not sure called flush with closeEvent = true twice is invalid.